test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/BodyModelBinderProviderTest.cs

```diff
@@ -70,11 +71,34 @@ public void GetBinder_WhenBindingSourceIsFromBody_ReturnsBinder()
             Assert.IsType<BodyModelBinder>(result);
         }
 
+        [Fact]
+        public void ObsoleteConstructor_DoesNotThrowNullReferenceException()
+        {
+            // Arrange
+            var context = new TestModelBinderProviderContext(typeof(Person));
+            context.BindingInfo.BindingSource = BindingSource.Body;
+            var formatter = new TestInputFormatter();
+            var formatterList = new List<IInputFormatter>();
```

Comment:
`var formatterList = new List<IInputFormatter> { formatter };` would be equivalent to the `.Add` on next line.